### PR TITLE
Clean up deprecated edm::RefToBase list 1 for RNTuple

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description:
This pull request is the first update to address the opened issue #42737 :

 ROOT is eventually deprecating the current TTree data storage in favor of their new `RNTuple`. The `RNTuple` is not planned to support dynamic polymorphism. One such use case in our data formats is `edm::RefToBase<T>` that internally uses dynamic-polymorphism based type erasure. The straightforward migration path is towards [`edm::Ptr<T>`](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMRef), but there are also many cases where the use of `edm::RefToBase<T>` could be cleaned up.

This PR covers the first case on the list in the issue #42737.
It contains the set of deprecated `edm::RefToBase` that only dictionary declaration and/or type aliases and/or commented-out code, otherwise unused case. 
All of these can be easily removed without any impact.

#### PR validation:
Tested in CMSSW_13_3_0_pre3, the basic test passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by rappoccio** [comment]
+1

File: DataFormats/BTauReco/src/classes_def.xml [hunk 1]
```diff
@@ -302,7 +302,6 @@
   <class name="std::vector<Measurement1D>"/>
 
   <!-- RefToBase stuff -->
-  <class name="edm::RefToBase<reco::BaseTagInfo>" />
   <class name="edm::reftobase::BaseHolder<reco::BaseTagInfo>" />
```
Comment (makortel) [inline DataFormats/BTauReco/src/classes_def.xml:305]: This PR was incomplete. All these `::reftobase::` dictionaries should be cleaned up as well. Could you do that?

**Pull request merged**